Support for Scan into pointer to RedisScan

Currently, calling `redis.Scan` with a destination that is a pointer to a type that implements `redis.RedisScan` results in the following error:
`redigo.Scan: cannot assign to dest 0: cannot convert from Redis simple string to *data.Foo`

Other flavors of Scan (e.g. `ScanStruct`) seem happy to `reflect.New` a target if necessary. Can the same be extended to `Scan`?

Example code:
```go
// scan_test.go
package data

import (
	"fmt"
	"strconv"
	"testing"

	"github.com/gomodule/redigo/redis"
	"github.com/stretchr/testify/assert"
)

type Foo struct{ Bar int64 }

func (f *Foo) RedisScan(src interface{}) error {
	switch s := src.(type) {
	case []byte:
		f.Bar, _ = strconv.ParseInt(string(s), 10, 64)
	case string:
		f.Bar, _ = strconv.ParseInt(string(s), 10, 64)
	default:
		return fmt.Errorf("invalid type %T", src)
	}
	return nil
}

func TestUnit_Scan(t *testing.T) {
	src := []interface{}{[]byte("1234"), nil}
	var foo *Foo
	_, err := redis.Scan(src, &foo)
	assert.NoError(t, err)
	assert.Equal(t, &Foo{Bar: 1234}, foo)
}
```

`go test` result:
```go

--- FAIL: TestUnit_Scan (0.00s)
    sandbox\scan_test.go:30: 
        	Error Trace:	scan_test.go:30
        	Error:      	Received unexpected error:
        	            	redigo.Scan: cannot assign to dest 0: cannot convert from Redis simple string to *data.Foo
        	Test:       	TestUnit_Scan
    sandbox\scan_test.go:31: 
        	Error Trace:	scan_test.go:31
        	Error:      	Not equal: 
        	            	expected: &data.Foo{Bar:1234}
        	            	actual  : (*data.Foo)(nil)
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,2 @@
        	            	-(*data.Foo)({
        	            	- Bar: (int64) 1234
        	            	-})
        	            	+(*data.Foo)(<nil>)
        	            	 
        	Test:       	TestUnit_Scan
FAIL
FAIL	sandbox	0.332s
Error: Tests failed.
```

--- a/redis/scan.go
+++ b/redis/scan.go
@@ -115,6 +115,26 @@ func convertAssignBulkString(d reflect.Value, s []byte) (err error) {
 		} else {
 			err = cannotConvert(d, s)
 		}
+	case reflect.Ptr:
+		if d.CanInterface() && d.CanSet() {
+			if s == nil {
+				if d.IsNil() {
+					return nil
+				}
+
+				d.Set(reflect.Zero(d.Type()))
+				return nil
+			}
+
+			if d.IsNil() {
+				d.Set(reflect.New(d.Type().Elem()))
+			}
+
+			if sc, ok := d.Interface().(Scanner); ok {
+				return sc.RedisScan(s)
+			}
+		}
+		err = convertAssignString(d, string(s))
 	default:
 		err = convertAssignString(d, string(s))
 	}

--- a/redis/scan_test.go
+++ b/redis/scan_test.go
@@ -18,10 +18,12 @@ import (
 	"fmt"
 	"math"
 	"reflect"
+	"strconv"
 	"testing"
 	"time"
 
 	"github.com/gomodule/redigo/redis"
+	"github.com/stretchr/testify/require"
 )
 
 type durationScan struct {
@@ -454,6 +456,68 @@ func TestArgs(t *testing.T) {
 		if !reflect.DeepEqual(tt.actual, tt.expected) {
 			t.Fatalf("%s is %v, want %v", tt.title, tt.actual, tt.expected)
 		}
+	}
+}
+
+type InnerStruct struct {
+	Foo int64
+}
+
+func (f *InnerStruct) RedisScan(src interface{}) (err error) {
+	switch s := src.(type) {
+	case []byte:
+		f.Foo, err = strconv.ParseInt(string(s), 10, 64)
+	case string:
+		f.Foo, err = strconv.ParseInt(s, 10, 64)
+	default:
+		return fmt.Errorf("invalid type %T", src)
+	}
+	return err
+}
+
+type OuterStruct struct {
+	Inner *InnerStruct
+}
+
+func TestScanPtrRedisScan(t *testing.T) {
+	tests := []struct {
+		name     string
+		src      []interface{}
+		dest     OuterStruct
+		expected OuterStruct
+	}{
+		{
+			name:     "value-to-nil",
+			src:      []interface{}{[]byte("1234"), nil},
+			dest:     OuterStruct{&InnerStruct{}},
+			expected: OuterStruct{Inner: &InnerStruct{Foo: 1234}},
+		},
+		{
+			name:     "nil-to-nil",
+			src:      []interface{}{[]byte(nil), nil},
+			dest:     OuterStruct{},
+			expected: OuterStruct{},
+		},
+		{
+			name:     "value-to-value",
+			src:      []interface{}{[]byte("1234"), nil},
+			dest:     OuterStruct{Inner: &InnerStruct{Foo: 5678}},
+			expected: OuterStruct{Inner: &InnerStruct{Foo: 1234}},
+		},
+		{
+			name:     "nil-to-value",
+			src:      []interface{}{[]byte(nil), nil},
+			dest:     OuterStruct{Inner: &InnerStruct{Foo: 1234}},
+			expected: OuterStruct{},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			_, err := redis.Scan(tc.src, &tc.dest.Inner)
+			require.NoError(t, err)
+			require.Equal(t, tc.expected, tc.dest)
+		})
 	}
 }
 

